File: samples/dark_castle.erl
% In a land shrouded in mystery...

-module(dark_castle).
-export([duration/0]).

duration() ->
    CastleName = "Dark Castle",
    ConstructionYear = 1100,
    DestructionYear = 1678,
    Duration = DestructionYear - ConstructionYear,
    io:format("The duration of ~s was ~p years.~n", [CastleName, Duration]).

% ...there stood a Dark Castle

File: samples/dark_castle.pas
{ In a land shrouded in mystery... }

program DarkCastle;

var
  CastleName: string;
  ConstructionYear, DestructionYear, Duration: integer;

begin
  CastleName := 'Dark Castle';
  ConstructionYear := 1100;
  DestructionYear := 1678;
  Duration := DestructionYear - ConstructionYear;
  writeln('The duration of ', CastleName, ' was ', Duration, ' years.');
end.

{ ...there stood a Dark Castle }
